app/src/main/java/woowacourse/shopping/feature/goods/GoodsViewModel.kt

```
@@ -3,40 +3,102 @@ package woowacourse.shopping.feature.goods
         page++
-        loadGoods()
+        loadItems()
+    }
+
+    fun insertToCart(cart: Cart) {
+        viewModelScope.launch {
+            try {
+                cartRepository.insert(cart)
+                updateItemsAndTotalQuantity(cart, cart.quantity + 1)
+                _isSuccess.setValue(Unit)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)